src/ThemeJsonExtractor.php

```diff
@@ -140,7 +142,20 @@ private static function read_json_file( $file_path ) {
 	 * @return array An array of theme.json fields that are translatable and the keys that are translatable.
 	 */
 	private static function get_fields_to_translate() {
-		$file_structure  = self::read_json_file( __DIR__ . '/theme-i18n.json' );
+		$context        = stream_context_create( array(
+			'http' => array(
+				'method'        => 'GET',
+				'header'        => 'Content-type: application/json',
+				'timeout'       => '3', // To make sure it resolves in a reasonable timeframe.
+			)
+		));
+		// Using the WordPress.org SVN repo resolved to a 403.
+		// $file_structure = self::read_json_file( 'http://develop.svn.wordpress.org/trunk/src/wp-includes/theme-i18n.json', $context );
+		$file_structure = self::read_json_file( 'https://raw.githubusercontent.com/WordPress/gutenberg/wp/trunk/lib/experimental-i18n-theme.json', $context );
```

Comment:
`https://raw.githubusercontent.com/WordPress/wordpress-develop/master/src/wp-includes/theme-i18n.json`?

Reply:
I've used the one in the `wp/trunk` branch because that's the one that only changes with WordPress core. I wish I could have used `wp/5.8` but that branch is not created until WordPress 5.8 is released like WordPress.org's SVN.

Reply:
So `https://raw.githubusercontent.com/WordPress/wordpress-develop/5.8/src/wp-includes/theme-i18n.json` perhaps?

Reply:
Locking it to 5.8 doesn't help at all, then it can just as well be statically committed here. The idea of an outside source is about getting the changes from Core without needing to release another WP-CLI version in sync. If you lock the version that WP-CLI looks for, there's not much of a point.

Reply:
It should be either:
1. always retrieve the translation info for the latest development version: or
2. always retrieve the translation info for the latest stable version.

Any other version is basically the same as having a hard-coded version inside of WP-CLI.

Reply:
I can change it to `master` in wordpress-develop.

However, as I shared above, making wp-cli dependant on an external file is fragile: there are changes to the `theme-i18n.json` that would work with the current code but there are other changes that won't.

Reply:
Done at https://github.com/wp-cli/i18n-command/pull/254/commits/eac73a112a7740b8bde17d82398041375d6dfd51

Reply:
Can you share more details about what type of changes wouldn't work and why can't we make them work?

Reply:
Any structural change to `theme.json` can affect this. A few ones I can think of right now: translatable strings need to be part of a leaf node that is an unkeyed array, the format supports variability in node names encoded via `*` but only to one level, can't translate several strings from the same leaf, etc.

It's not that these are planned changes. My point is that, in the current state of things, the unit that best encapsulates the behavior is the file + the code used to parse it. Hence, pulling the file from elsewhere is risky: can work with some changes, and won't work with others.

Reply:
hi @schlessera is there anything I can help with to move this forward?

Reply:
Not at the moment. I'll try to find the time to do a thorough final review, and then it will be good to be merged.
What I'll be looking at most of all, is ways in which this can break if the format changes, and ensure we have graceful handling for such cases (instead of causing a fatal or - even worse - succeeding with the wrong output).

Reply:
Feel free to push changes to this PR if you want to. I'm also happy to add more graceful handling if we come up with anything.